NavHost cannot be null on Back key pressed

### Description

I noticed InvalidOperationException (NavHost cannot be null) when running app with debugger on Android 12 on Back key pressed.

### Steps to Reproduce

1. Create Maui app
2. Define MainPage in App.xaml.cs as:
```
var rootPage = new MainPage();
MainPage = new NavigationPage(rootPage);
```
3. Run app with debugger on Android 12 
4. Move app to background using Back key
5. Return to the app using main launcher (icon)
6. Press Back key again

Actual result:
InvalidOperationException is thrown:

```
NavHost cannot be null

   at Microsoft.Maui.Platform.StackNavigationManager.get_NavHost() in D:\a\_work\1\s\src\Core\src\Platform\Android\Navigation\StackNavigationManager.cs:line 32
   at Microsoft.Maui.Platform.NavigationViewFragment.OnResume() in D:\a\_work\1\s\src\Core\src\Platform\Android\Navigation\NavigationViewFragment.cs:line 75
   at AndroidX.Fragment.App.Fragment.n_OnResume(IntPtr jnienv, IntPtr native__this) in /Users/runner/work/1/s/generated/androidx.fragment.fragment/obj/Release/net6.0-android/generated/src/AndroidX.Fragment.App.Fragment.cs:line 2614
   at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PP_V(_JniMarshal_PP_V callback, IntPtr jnienv, IntPtr klazz) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:line 22
```

Expected result:
No exception

### Link to public reproduction project repository

https://github.com/luwed/maui-stuff/tree/NavHost-cannot-be-null-on-back

### Version with bug

Unknown/Other (please specify)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 12

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

Checked with Visual Studio Enterprise 2022 for Mac Version 17.6.8 and Microsoft Visual Studio Community 2022 (64-bit) Version 17.8.4. Results (with info about operating system) below.

- macOS Sonoma
  - Visual Studio Enterprise 2022 for Mac Version 17.6.8 (build 400)/.NET 7.0 - issue can be reproduced
  - Visual Studio Enterprise 2022 for Mac Version 17.6.8 (build 400)/project converted to .NET 8.0 - different issue occurred, stacktrace below

`
[ViewRootImpl@229e1d[MainActivity]] stopped(true) old=false
[mono-rt] [ERROR] FATAL UNHANDLED EXCEPTION: Java.Lang.IllegalStateException: Can not perform this action after onSaveInstanceState
[mono-rt]    at Java.Interop.JniEnvironment.InstanceMethods.CallIntMethod(JniObjectReference instance, JniMethodInfo method, JniArgumentValue* args) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/obj/Release/net7.0/JniEnvironment.g.cs:line 20203
[mono-rt]    at Java.Interop.JniPeerMembers.JniInstanceMethods.InvokeAbstractInt32Method(String encodedMember, IJavaPeerable self, JniArgumentValue* parameters) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/Java.Interop/JniPeerMembers.JniInstanceMethods_Invoke.cs:line 492
[mono-rt]    at AndroidX.Fragment.App.BackStackRecord.Commit() in C:\a\_work\1\s\generated\androidx.fragment.fragment\obj\Release\net6.0-android\generated\src\AndroidX.Fragment.App.BackStackRecord.cs:line 131
[mono-rt]    at Microsoft.Maui.Platform.NavigationRootManager.SetContentView(IView view) in D:\a\_work\1\s\src\Core\src\Platform\Android\Navigation\NavigationRootManager.cs:line 129
[mono-rt]    at Microsoft.Maui.Platform.NavigationRootManager.Disconnect() in D:\a\_work\1\s\src\Core\src\Platform\Android\Navigation\NavigationRootManager.cs:line 113
[mono-rt]    at Microsoft.Maui.Handlers.WindowHandler.DisconnectHandler(NavigationRootManager navigationRootManager) in D:\a\_work\1\s\src\Core\src\Handlers\Window\WindowHandler.Android.cs:line 83
[mono-rt]    at Microsoft.Maui.Handlers.WindowHandler.OnDisconnectHandler(Object platformView) in D:\a\_work\1\s\src\Core\src\Handlers\Window\WindowHandler.Android.cs:line 67
[mono-rt]    at Microsoft.Maui.Handlers.ElementHandler.DisconnectHandler(Object platformView) in D:\a\_work\1\s\src\Core\src\Handlers\Element\ElementHandler.cs:line 112
[mono-rt]    at Microsoft.Maui.Handlers.ElementHandler.Microsoft.Maui.IElementHandler.DisconnectHandler() in D:\a\_work\1\s\src\Core\src\Handlers\Element\ElementHandler.cs:line 130
[mono-rt]    at Microsoft.Maui.Controls.Element.SetHandler(IElementHandler newHandler) in D:\a\_work\1\s\src\Controls\src\Core\Element\Element.cs:line 918
[mono-rt]    at Microsoft.Maui.Controls.Element.set_Handler(IElementHandler value) in D:\a\_work\1\s\src\Controls\src\Core\Element\Element.cs:line 863
[mono-rt]    at Microsoft.Maui.Platform.ElementExtensions.SetHandler(Context nativeElement, IElement element, IMauiContext context) in D:\a\_work\1\s\src\Core\src\Platform\ElementExtensions.cs:line 156
[mono-rt]    at Microsoft.Maui.Platform.ElementExtensions.SetWindowHandler(Activity platformWindow, IWindow window, IMauiContext context) in D:\a\_work\1\s\src\Core\src\Platform\ElementExtensions.cs:line 166
[mono-rt]    at Microsoft.Maui.Platform.ApplicationExtensions.CreatePlatformWindow(Activity activity, IApplication application, Bundle savedInstanceState) in D:\a\_work\1\s\src\Core\src\Platform\Android\ApplicationExtensions.cs:line 48
[mono-rt]    at Microsoft.Maui.MauiAppCompatActivity.OnCreate(Bundle savedInstanceState) in D:\a\_work\1\s\src\Core\src\Platform\Android\MauiAppCompatActivity.cs:line 35
[mono-rt]    at Android.App.Activity.n_OnCreate_Landroid_os_Bundle_(IntPtr jnienv, IntPtr native__this, IntPtr native_savedInstanceState) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net8.0/android-34/mcw/Android.App.Activity.cs:line 3082
[mono-rt]    at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PPL_V(_JniMarshal_PPL_V callback, IntPtr jnienv, IntPtr klazz, IntPtr p0) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:line 125
[mono-rt]   --- End of managed Java.Lang.IllegalStateException stack trace ---
[mono-rt] java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
[mono-rt] 	at androidx.fragment.app.FragmentManager.checkStateLoss(FragmentManager.java:1632)
[mono-rt] 	at androidx.fragment.app.FragmentManager.enqueueAction(FragmentManager.java:1672)
[mono-rt] 	at androidx.fragment.app.BackStackRecord.commitInternal(BackStackRecord.java:341)
[mono-rt] 	at androidx.fragment.app.BackStackRecord.commit(BackStackRecord.java:306)
[mono-rt] 	at crc6488302ad6e9e4df1a.MauiAppCompatActivity.n_onCreate(Native Method)
[mono-rt] 	at crc6488302ad6e9e4df1a.MauiAppCompatActivity.onCreate(MauiAppCompatActivity.java:49)
[mono-rt] 	at android.app.Activity.performCreate(Activity.java:8290)
[mono-rt] 	at android.app.Activity.performCreate(Activity.java:8270)
[mono-rt] 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
[mono-rt] 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4085)
[mono-rt] 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4277)
[mono-rt] 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
[mono-rt] 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
[mono-rt] 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
[mono-rt] 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2443)
[mono-rt] 	at android.os.Handler.dispatchMessage(Handler.java:106)
[mono-rt] 	at android.os.Looper.loopOnce(Looper.java:226)
[mono-rt] 	at android.os.Looper.loop(Looper.java:313)
[mono-rt] 	at android.app.ActivityThread.main(ActivityThread.java:8751)
[mono-rt] 	at java.lang.reflect.Method.invoke(Native Method)
[mono-rt] 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
[mono-rt] 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
[mono-rt] 
[mono-rt]   --- End of managed Java.Lang.IllegalStateException stack trace ---
[mono-rt] java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
[mono-rt] 	at androidx.fragment.app.FragmentManager.checkStateLoss(FragmentManager.java:1632)
[mono-rt] 	at androidx.fragment.app.FragmentManager.enqueueAction(FragmentManager.java:1672)
[mono-rt] 	at androidx.fragment.app.BackStackRecord.commitInternal(BackStackRecord.java:341)
[mono-rt] 	at androidx.fragment.app.BackStackRecord.commit(BackStackRecord.java:306)
[mono-rt] 	at crc6488302ad6e9e4df1a.MauiAppCompatActivity.n_onCreate(Native Method)
[mono-rt] 	at crc6488302ad6e9e4df1a.MauiAppCompatActivity.onCreate(MauiAppCompatActivity.java:49)
[mono-rt] 	at android.app.Activity.performCreate(Activity.java:8290)
[mono-rt] 	at android.app.Activity.performCreate(Activity.java:8270)
[mono-rt] 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
[mono-rt] 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4085)
[mono-rt] 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4277)
[mono-rt] 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
[mono-rt] 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
[mono-rt] 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
[mono-rt] 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2443)
[mono-rt] 	at android.os.Handler.dispatchMessage(Handler.java:106)
[mono-rt] 	at android.os.Looper.loopOnce(Looper.java:226)
[mono-rt] 	at android.os.Looper.loop(Looper.java:313)
[mono-rt] 	at android.app.ActivityThread.main(ActivityThread.java:8751)
[mono-rt] 	at java.lang.reflect.Method.invoke(Native Method)
[mono-rt] 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:571)
[mono-rt] 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
[mono-rt] 
`
- Windows 11
  - Microsoft Visual Studio Community 2022 (64-bit) Version 17.8.4/.NET 7.0 - issue can't be reproduced
  - Microsoft Visual Studio Community 2022 (64-bit) Version 17.8.4/project converted to .NET 8.0 - issue can't be reproduced

--- a/src/Core/src/Platform/Android/Navigation/NavigationViewFragment.cs
+++ b/src/Core/src/Platform/Android/Navigation/NavigationViewFragment.cs
@@ -69,7 +69,7 @@ namespace Microsoft.Maui.Platform
 
 		public override void OnResume()
 		{
-			if (_currentView == null || NavigationManager.NavHost == null)
+			if (_currentView == null || !NavigationManager.HasNavHost)
 				return;
 
 			if (_currentView.Parent == null)

--- a/src/Core/src/Platform/Android/Navigation/StackNavigationManager.cs
+++ b/src/Core/src/Platform/Android/Navigation/StackNavigationManager.cs
@@ -46,6 +46,8 @@ namespace Microsoft.Maui.Platform
 		internal NavGraph NavGraph => _navGraph ??
 			throw new InvalidOperationException($"NavGraph cannot be null");
 
+		internal bool HasNavHost => _navHost is not null;
+
 		public IView CurrentPage
 			=> _currentPage ?? throw new InvalidOperationException("CurrentPage cannot be null");
 

